src/Module/Profile.php

```diff
@@ -54,15 +54,28 @@ public static function rawContent()
 	{
 		if (ActivityPub::isRequest()) {
 			$user = DBA::selectFirst('user', ['uid'], ['nickname' => self::$which]);
+			$data = [];
 			if (DBA::isResult($user)) {
 				$data = ActivityPub\Transmitter::getProfile($user['uid']);
+			}
+
+			if (!empty($data)) {
 				System::jsonExit($data, 'application/activity+json');
 			} elseif (DBA::exists('userd', ['username' => self::$which])) {
 				// Known deleted user
-				System::httpExit(410);
+				$data = [
+					'@context' => ActivityPub::CONTEXT,
+					'id' => self::getApp()->getBaseUrl() . '/profile/' . self::$which,
+					'type' => 'Tombstone',
+					'published' => DateTimeFormat::utcNow(DateTimeFormat::ATOM),
+					'updated' => DateTimeFormat::utcNow(DateTimeFormat::ATOM),
+					'deleted' => DateTimeFormat::utcNow(DateTimeFormat::ATOM),

```

Comment:
I would suggest to add the deletion date to the "userd" table and to use it here. Additionally I would like to check if this is totally AP conform.

Reply:
I copied the structure outlined here: https://www.w3.org/TR/activitypub/#delete-activity-outbox

Adding the deletion date to the userd table is good but I won't add it to this PR myself.

Reply:
Nonetheless I think that this has to part of the "Transmitter" class of the ActivityPub protocol.

Reply:
Fair enough, I can move this code away from the Module.